exporter/exporter.go

```diff
@@ -117,26 +125,41 @@ func (e *Exporter) makeRegistry(ctx context.Context, client *mongo.Client, topol
 	}
 	registry.MustRegister(&gc)
 
+	if client == nil {
+		return registry
+	}
+
 	nodeType, err := getNodeType(ctx, client)
 	if err != nil {
 		e.logger.Errorf("Cannot get node type to check if this is a mongos: %s", err)
 	}
 
-	// enable collection dependant collectors like collstats and indexstats
-	enableCollStats := false
-	if e.opts.CollStatsLimit == -1 {
-		enableCollStats = true
-	}
+	// enable collectors like collstats and indexstats depending on the number of collections
+	// present in the database.
+	limitsOk := false
 	if e.getTotalCollectionsCount() > 0 && e.getTotalCollectionsCount() < e.opts.CollStatsLimit {
-		enableCollStats = true
+		limitsOk = true
+	}
+	if e.opts.CollStatsLimit == -1 { // -1: unlimited
+		limitsOk = true
+	}
+
+	if e.opts.CollectAll {
+		if len(e.opts.CollStatsNamespaces) == 0 {
+			e.opts.DiscoveringMode = true
+		}
+		e.opts.EnableDiagnosticData = true
+		e.opts.EnableDBStats = true
+		e.opts.CollectorTopMetrics = true
+		e.opts.EnableReplicasetStatus = true
 	}
 
 	// if we manually set the collection names we want or auto discovery is set
-	if (len(e.opts.CollStatsCollections) > 0 || e.opts.DiscoveringMode) && enableCollStats {
+	if (len(e.opts.CollStatsNamespaces) > 0 || e.opts.DiscoveringMode) && e.opts.EnableCollStats && limitsOk {
```

Comment:
it looks weird that when user just adds `--collector.collstats` without limits it doesn't enable it. 
I think in exporter if user doesn't provide limit and calls --collector.collstats` it should be unlimited.
The same for `collector.indexstats.`
@rnovikovP @denisok your thoughts?

Reply:
should we shift that logic to pmm? it should check limits. For exporter I think you are right, but chances that someone shoots himself in the leg are high.
But yeah, I would do the logic in pmm and have it set limit flags to disabled and leave it here enabled by default.

Reply:
I don't mean that we should enable collstats by default I mean that user should be able to enable it just by calling one flag instead of two.
How it works now
`mongodb_expoter --collector.collstats --collector.collstats-limit=-1`
How it should work
`mongodb_expoter --collector.collstats`

So if you don't pass collstats limit collstats collector won't be enabled.